src/App.tsx

```diff
@@ -1,26 +1,141 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoApi from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [status, setStatus] = useState(Status.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processedId, setProcessedId] = useState<number[]>([]);
+
+  const errorTimer = useRef(0);
+
+  const handleErrorMessage = (message: string) => {
+    setErrorMessage(message);
+
+    window.clearTimeout(errorTimer.current);
+
+    errorTimer.current = window.setTimeout(() => setErrorMessage(''), 3000);
+  };
+
+  useEffect(() => {
+    todoApi
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    if (status === Status.all) {
+      return todos;
+    }
+
+    return todos.filter(todo => {
+      return status === Status.completed ? todo.completed : !todo.completed;
+    });
+  }, [todos, status]);
+
+  function addTodo({ title, userId, completed }: Todo) {
+    handleErrorMessage('');
+    setTempTodo({ title, userId, completed, id: 0 });
+
+    return todoApi
+      .createTodos({ title, userId, completed })
+      .then(todoToAdd => {
+        setTodos(currentTodos => [...currentTodos, todoToAdd]);
+        setTempTodo(null);
+      })
+      .catch(error => {
+        setTempTodo(null);
+        handleErrorMessage('Unable to add a todo');
+        throw error;
+      });
+  }
+
+  function deleteTodo(id: number) {
```

Comment:
It is important to follow the same way of creating functions everywhere. Fix it everywhere
```suggestion
  const deleteTodo = (id: number) => {
```